mobs/nakki_v2/NakkiV2.cs

```diff
@@ -0,0 +1,291 @@
+using Godot;
+using System;
+using System.Linq;
+
+using Jakojaannos.WisperingWoods.Characters.Player;
+using Jakojaannos.WisperingWoods.Util.Editor;
+
+namespace Jakojaannos.WisperingWoods;
+
+[Tool]
+public partial class NakkiV2 : Path2D {
+	public override string[] _GetConfigurationWarnings() {
+		var warnings = base._GetConfigurationWarnings() ?? Array.Empty<string>();
+
+		if (Curve == null) {
+			warnings = warnings.Append("Add a curve for näkki to move along (under 'Path2D' in inspector)").ToArray();
+		} else if (Curve.PointCount == 0) {
+			warnings = warnings.Append("Add points for näkki's movement path (Curve resource under 'Path2D' in inspector)").ToArray();
+		}
+
+		if (_defaultState == null) {
+			warnings = warnings.Append("Default state is not set").ToArray();
+		}
+
+		return warnings
+			.Union(this.CheckCommonConfigurationWarnings())
+			.ToArray();
+	}
+
+	[Export] private float _speed = 50.0f;
+	[Export] private NakkiAiState? _defaultState;
+
+	private NakkiAiState? _currentState;
+	private NakkiAiState? CurrentState {
+		get => _currentState;
+		set {
+			CurrentState?.ExitState(this);
+			_currentState = value;
+			CurrentState?.EnterState(this);
+		}
+	}
+
+	public PathFollow2D? _nakkiEntity;
+	private float? _targetProgress;
+
+	private Player? _player;
+	private RayCast2D? _lineOfSight;
+	public float _detectionLevel = 0.0f;
+	[Export] private float _detectionGain = 100.0f;
+	[Export] private float _detectionDecay = 60.0f;
```

Comment:
Why are these private fields? Could these be updated to public properties? As I said earlier when we discussed the `BindingFlags` thing, I see any `[Export]` as inherently `public`. The only exception I can think of is something that breaks if updated at runtime (so you'd like that to be configurable in-editor but not at runtime.

Looking at code, modifying these from code at runtime is (while a bit silly) completely safe, so I'd just change these to public props.
```suggestion
	[Export]
	private float DetectionGain { get; set; } = 100.0f;
	[Export]
	public float DetectionDecay { get; set; } = 60.0f;
```